doc/decisions/dr-004.md

```diff
@@ -0,0 +1,178 @@
+  - Pros:
+    - Compared to option 1a), only a fraction of the adapters would need to be written.
+  - Cons:
+    - Our board adapters would have a high degree of dependency on the ACA.
+      The *Single Responsibility Principle* is violated[²](<\ref dr004_2 "²") as the board adapters would not only depend on the board and its hardware but also on an interface of a 3rd party HAL.
+    - Dealing with the extensions made compared to vanilla ACA requires writing some adapters.
+    - Adapting a variation from the ACA in one member function of a class requires creating an interface containing all used member functions of that class, even if those other member functions are used without alterations.
+      This introduces *boilerplate code*.
+- **Option 2b)**: Use alternative sections within source files
+  - Pros:
+    - Reduces the need for additional code to a minimum.
+      No adapters need to be written to use the HAL.
+  - Cons:
+    - Our board adapters would have a high degree of dependency on the ACA.
+      The *Single Responsibility Principle* is violated[²](<\ref dr004_2 "²") even further (compared to option 2a)) by depending on the vanilla ACA required for testing in the native build environment only.
+      Changes to the test infrastructure should not necessitate changes in the source file containing productive code.
+    - In general, conditionally compiled sections of code reduces its readability.
+
+\note \anchor dr004_1
+¹: *In practice*, our HAL interface may not be suitable to accommodate the new HAL implementation (see [Rule of Three](https://en.wikipedia.org/w/index.php%3Ftitle%3DRule_of_three_%28computer_programming%29%26oldid%3D1173684754)).
+Thus, we would have to adjust our HAL interface and the code using it accordingly.
+In other words, it is difficult to design a HAL interface such that we would actually benefit when changing the HAL implementation to a different 3rd party library.
+Thus, this advantage may not be relevant for that case.  
+Also, in case the HAL implementation is changed, it may be less effort to adjust all the uses of the new HAL interface than to write adapters for the changed 3rd party library.
+
+\note \anchor dr004_2
+²: The source file (of a non-adapter) would depend on a 3rd party interface.
+Changes to that 3rd party interface may necessitate changes to the source file even though the requirements to that source file remain the same.
+
+### Summary & Decision
+
+Writing adapters adds additional effort, potentially resulting in boilerplate code, and the advantage is dubious.
+
+After careful consideration, we've decided on:
+
+> **Option 2:** Allow direct dependencies from hardware-related code to 3rd party HAL interfaces.
+
+And more specifically for testing on native environments, use:
+
+> **Option 2b):** Where the ACA is used, it is allowed to add small conditionally compiled sections for tests to cope with deviations of the ACA used in productive code from the vanilla ACA.
```

Comment:
This is not what I [discussed](https://github.com/Task-Tracker-Systems/.github/wiki/Meeting-notes-2024%E2%80%9002%E2%80%9013#dependency-to-arduinocore-105) with @FiveTeethless on 2024-02-13. Originally we wanted to settle on option 2a). After documenting this decision in detail I found that option 2b) may be better. I am *open for comments* and may be convinced otherwise!